app/js/backbone-views.js

```diff
@@ -1694,56 +1694,105 @@ var PromptInvalidURLWarning = Backbone.View.extend({
+    }else if(type == "left"){
+      $('#template-reversible-input-table :input.template-reaction-textbox[name="'+i+'"]').closest('tr').remove();
+    }else{
+      $('#template-reversible-output-table :input.template-reaction-textbox[name="'+i+'"]').closest('tr').remove();
+    }
   },
-  switchInputOutput: function () {
-    var saveHtmlContent = $("#reaction-template-left-td").html();
-    $("#reaction-template-left-td").html($("#reaction-template-right-td").html());
-    $("#reaction-template-right-td").html(saveHtmlContent);
+  switchInputOutput: function (e) {
+    var self = this;
+    if(e == "association") {
+      $('#reaction-template-left-td').html(self.associatedHTMLContent);
+      $('#reaction-template-right-td').html(self.dissociatedHTMLContent);
+    }else if(e == "dissociation"){
+      $('#reaction-template-left-td').html(self.dissociatedHTMLContent);
+      $('#reaction-template-right-td').html(self.associatedHTMLContent);
+    }else{
+      $('#reaction-template-left-td').html(self.reversibleInputHTMLContent);
+      $('#reaction-template-right-td').html(self.reversibleOutputHTMLContent);
+      self.disableDeleteButtonStyle("left");
+      self.disableDeleteButtonStyle("right");
+    }
   },
   getAllParameters: function () {
     var templateType = $('#reaction-template-type-select').val();
     var templateReactionComplexName = $('#template-reaction-complex-name').val();
     var macromoleculeList = $('#template-reaction-dissociated-table :input.template-reaction-textbox').map(function(){
         return $(this).val()
       }).toArray();
+    var reversibleInputMacromoleculeList = $('#template-reversible-input-table :input.template-reaction-textbox').map(function(){
+      return $(this).val();
+    }).toArray();
+    var reversibleOutputMacromoleculeList = $('#template-reversible-output-table :input.template-reaction-textbox').map(function(){
+      return $(this).val();
+    }).toArray();
     // enable complex name only if the user provided something
     var templateReactionEnableComplexName = $.trim(templateReactionComplexName).length != 0;
 
     return {
       templateType: templateType,
       templateReactionComplexName: templateReactionComplexName,
       macromoleculeList: macromoleculeList,
+      reversibleInputMacromoleculeList: reversibleInputMacromoleculeList,
+      reversibleOutputMacromoleculeList: reversibleOutputMacromoleculeList,
       templateReactionEnableComplexName: templateReactionEnableComplexName
     }
   },
-  disableDeleteButtonStyle: function () {
-    $("img.template-reaction-delete-button").css("opacity", 0.2);
-    $("img.template-reaction-delete-button").css("cursor", "default");
+  disableDeleteButtonStyle: function (type) {
+    if(type == "reaction"){
+      $("img.template-reaction-delete-button").css("opacity", 0.2);
+      $("img.template-reaction-delete-button").css("cursor", "default");
+    }else if(type == "left"){
```

Comment:
Let's follow the format of
`if ( ) {

}
else if {

}`
here and a few more places since it is generally the case.